docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2019-abelbriggs.md

```diff
@@ -0,0 +1,256 @@
+`.shader_test` files, render the included shader programs, test images for correct color values,
+ and more.
+ 
+For running shaders, GraphicsFuzz uses a client-server model. In GraphicsFuzz's toolset, a webserver
+and several 'workers', or clients, are supplied. These workers and the server communicate to each
+other via an [Apache Thrift](https://thrift.apache.org/) specification.
+ 
+To run a test set, the user starts the webserver and a worker, then selects a test set to run 
+(via UI or otherwise). For each shader, the webserver sends the shader source code and a JSON file
+that details any additional information the shader needs (such as uniform data) to the worker - the
+worker then renders the shader in any way it sees fit and returns the rendered image
+(if there was one), logs, etc.
+
+When reporting bugs in Mesa's drivers, it is most convenient for the Mesa developers to have a
+`.shader_test` file that they can easily plug into Piglit as a test case. Additionally, the OpenGL ES
+GraphicsFuzz worker is rather unwieldy to compile/use for desktop platforms, and so it was decided
+that the development of a Piglit/`shader_runner` worker was worth pursuing, along with a script to
+systematically convert GraphicsFuzz shader jobs into Piglit `.shader_test` files.
+
+The following PRs involve the Piglit converter script or worker.
+
+[#577](https://github.com/google/graphicsfuzz/pull/577):
+Add script to convert a GraphicsFuzz shader job to a piglit shader test
+
+[#584](https://github.com/google/graphicsfuzz/pull/584):
+Fix piglit converter script not inserting GL ES version
+
+[#587](https://github.com/google/graphicsfuzz/pull/587):
+Refactor common python script functions into one common lib
+
+[#613](https://github.com/google/graphicsfuzz/pull/613):
+Fix piglit converter not accepting blank JSON
+
+[#617](https://github.com/google/graphicsfuzz/pull/617):
+Add piglit shader_runner_gles3 worker script
+
+[#631](https://github.com/google/graphicsfuzz/pull/631):
+docs: Add documentation for piglit worker
+
+[#632](https://github.com/google/graphicsfuzz/pull/632):
+Piglit worker improvements
+
+[#638](https://github.com/google/graphicsfuzz/pull/638):
+graphicsfuzz-piglit-converter: support more uniform types
+
+In addition, a contribution was made to Piglit upstream to allow `shader_runner` to handle a case
+when unused uniform data was optimized out of a compiled shader.
+
+[shader_runner: Add command line option to ignore uniform if it isn't found in shader](https://gitlab.freedesktop.org/mesa/piglit/merge_requests/92)
+
+#### Leftovers
+
+The following issues are nits or minor problems related to the Piglit worker or converter script
+that were left unfixed due to time or knowledge constraints.
+
+[#597](https://github.com/google/graphicsfuzz/issues/597):
+Support compute shaders in graphicsfuzz_piglit_converter
+
+### Add new shaders to GraphicsFuzz's test set
```

Comment:
It would be nice to show the images for these shaders.

Reply:
- Should I include images for the integer variants if they're the same as the standard ones?
 - Should I put the images into the images directory or upload to a image host? (Asking since this doc will be copypasta'd into a gist for submittal.)

Reply:
I was unable to put the images in the images directory, since this report is in a new directory under summer-of-code-ideas. So I just uploaded to imgur instead.